src/main/java/com/google/sps/servlets/getData.java

```diff
@@ -55,133 +45,91 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
-      
-   // Fill the hashmap with our data! (Read our data line by line, splitting by comma)
-      while(input.hasNextLine()) {
-         String[] tokens = input.nextLine().split(",");
-         
-         String county = tokens[1].toLowerCase();
-         String state  = tokens[2].toLowerCase();
-         
-         // DEBUG: System.out.println("State: " + state + " County: " + county + " Cases: " + tokens[4] + " Deaths: " + tokens[5]);
-         
-         int cases  = tokens[4].isEmpty() ? 0 : Integer.parseInt(tokens[4]);
-         int deaths = tokens[5].isEmpty() ? 0 : Integer.parseInt(tokens[5]);
-         
-         Pair<String, String> myCounty = new Pair<String, String>(state, county);
-         Pair<Integer, Integer> myCountyData = new Pair<Integer, Integer>(cases, deaths);
-         
-         dataMap.put(myCounty, myCountyData);
+      input.readLine();
+   
+   // The file contains a month's worth of information, with the most recent being yesterday's data.
+   // Save yesterday's date, which we can use to tell the computer when to stop
+      String yesterday = LocalDate.now().minusDays(1).toString();
+
+   // Read a single line from our file, and initialize our tokens and fileDay (the day the file is currently on)
+      String[] tokens = input.readLine().split(",");
+      String fileDay = tokens[0];
+
+   // Skip through the file line by line until we reach yesterday's date. 
+      while(!fileDay.equals(yesterday)) {
+          tokens = input.readLine().split(",");
+          fileDay = tokens[0];
+      }
+
+   // Now that we've reached yesterday's date, scan through the rest of the file for the record containing our state and county.
+
+   // Desired state and county
+      String state  = request.getParameter("state").toLowerCase();
+      String county = request.getParameter("county").toLowerCase();
+
+   // State and County the file is currently on
+      String fileState  = tokens[3].toLowerCase();
+      String fileCounty = tokens[2].toLowerCase();
+
+      String[] results = null;
+      String line;
+
+   // Loop through the rest of the file. if we reach EOF, that means location not found.
+      while((line = input.readLine()) != null) {
+
+     // If the line we're on has the same county and state we're looking for...
+        if(fileState.equals(state) && fileCounty.equals(county))
+            results = tokens; // line found!
+        else {
+            tokens = line.split(",");
+            fileState  = tokens[3].toLowerCase();
+            fileCounty = tokens[2].toLowerCase();
+        }
       }
       
       input.close();
 
 
-// #############################
-// # Displaying Output To User #  In this stage, we create JSON from all the data we've collected.  
-// #############################
+// #######################
+// # Output Data To JSON #  In this stage, we create JSON from all the data we've collected.  
+// #######################
 
       response.setContentType("application/json;");
 
```

Comment:
Again, starting here, most of the changes are removing old code that isn't necessary anymore- for example, since there's no more hashmap, no need for all that hashmap code! Same with the Pair class, no longer needed!